### PR TITLE
[Impeller] Fix swapchain recreation for non-polling cases.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Earlier, there was no recovery from vk::Result::eSuboptimalKHR. This made the playground be stuck with a swapchain image size of 1x1 with suboptimal error code on acquisition indefinitely.

Towards fixing the playgrounds. There are still some validation errors.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
If I am following this correctly, the polling on Android took care of this case on that platform.

**Review by matanlurey — COMMENTED** [review]
LGTM but defer to Jonah for explicit approval since this is my second time in this file :)

**Review by jonahwilliams — APPROVED** [review]
LGTM, if it works it works.

**Pull request merged**
